src/main/com/diana/restaurant/queue/QueueManager.java

```diff
@@ -0,0 +1,33 @@
+package com.diana.restaurant.queue;
+
+import com.diana.restaurant.entities.Order;
+import com.diana.restaurant.enums.Queues;
+import com.diana.restaurant.exceptions.queueManager.QueueManagerKeyNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+public class QueueManager {
+  private static QueueManager instance;
+  private final Map<String, Object> instanceMap = new HashMap<>();
+
+  private QueueManager() {
+    this.registerQueues(this.instanceMap);
+  }
+
+  private void registerQueues(Map<String, Object> instanceMap) {
+    instanceMap.put(Queues.CHECK_INGREDIENTS_QUEUE, new ConcurrentLinkedQueue<Order>());
+  }
+
+  public static QueueManager getInstance() {
+    return Optional.ofNullable(instance).orElseGet(QueueManager::new);
+  }
+
+  public <T> T get(String key) {
```

Comment:
You should not return the queue object, let queue manager handle all the operations with any queue

Reply:
Added enQueue and deQueue methods in queueManager